client/my-sites/media/main.jsx

```diff
@@ -44,14 +52,91 @@ module.exports = React.createClass( {
 		page( redirect );
 	},
 
+	openDetailsModal( item ) {
+		this.setState( { openedDetails: item } );
+	},
+
+	closeDetailsModal() {
+		this.setState( { openedDetails: null, editedItem: null } );
+	},
+
+	editImage() {
+		this.setState( { openedDetails: null, editedItem: this.state.openedDetails } );
+	},
+
+	onImageEditorCancel: function( imageEditorProps ) {
```

Comment:
I agree with you, but that's an issue with ImageEditor design and I'd like not to take it on in this PR :)